sdk/core/core/src/az_http_header_validation_private.h

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file az_http_header_validation_private.h
+ *
+ * @brief This header defines a bit array that is used to validate whenever or not an ASCII char is
+ * valid within an http header name.
+ *
+ */
+
+#ifndef _az_HTTP_HEADER_VALIDATION_PRIVATE_H
+#define _az_HTTP_HEADER_VALIDATION_PRIVATE_H
+
+#include <az_span.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <_az_cfg_prefix.h>
+
+// Bit array from ASCII to valid. Every zero in array means invalid character, while non-zeros
+// return the valid character.
+static const uint8_t az_http_valid_token[256] = {
```

Comment:
nit: consider representing this in a compact 16x16 grid where we just have 0/1 binary values rather than 0 and non-zero.

Reply:
our auto-formatter didn't like 16x16.  it takes control over! :)
That's one of the motivation of having this as an individual header.  C-world people is used to long files, I'd say.

I used non-zero and the value instead in casa we want to reuse the same array to lower-Case or something like that.  There's not really any significant benefit from one of other